specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json

```diff
@@ -10539,7 +10644,187 @@
+        "metaData": {
+          "$ref": "#/definitions/GetInsightsResultsMetadata",
+          "description": "The metadata from the get insights operation results."
+        },
+        "value": {
+          "description": "The insights result values.",
+          "items": {
+            "$ref": "#/definitions/EntityInsightItem"
+          },
+          "type": "array"
+        }
+      }
+    },
+    "GetInsightsResultsMetadata": {
+      "description": "Get Insights result metadata.",
+      "properties": {
+        "totalCount": {
+          "description": "the total items found for the insights request",
+          "type": "integer",
+          "format": "int32"
+        },
+        "errors": {
+          "description": "information about the failed queries",
+          "items": {
+            "$ref": "#/definitions/GetInsightsError"
+          },
+          "type": "array"
+        }
+      },
+      "required": [
+        "totalCount",
+        "aggregations"
+      ],
+      "type": "object"
+    },
+    "GetInsightsError": {
+      "description": "GetInsights Query Errors.",
+      "properties": {
+        "kind": {
+          "description": "the query kind",
+          "type": "string",
+          "enum": ["Insight"]
+        },
+        "queryId": {
+          "description": "the query id",
+          "type": "string"
+        },
+        "errorMessage": {
+          "description": "the error message",
+          "type": "string"
+        }
+      },
+      "required": [
+        "kind",
+        "errorMessage"
+      ],
+      "type": "object"
+    },
+    "EntityInsightItem": {
+      "description": "Entity insight Item.",
+      "type": "object",
+      "properties": {
+        "queryId": {
+          "type": "string",
+          "description": "The query id of the insight"
+        },
+        "queryTimeInterval": {
+          "type": "object",
+          "description": "The Time interval that the query actually executed on.",
+              "properties": {
+                "startTime":{
+                    "format": "date-time",
+              "type": "string",
+              "description": "Insight query start time"
+            },
+            "endTime":{
+              "format": "date-time",
+              "type": "string",
+              "description": "Insight query end time"
+            }
+            
+          }
+        },
+        "tableQueryResults": {
+          "$ref": "#/definitions/InsightsTableResult",
+          "description": Query results for table insights query."
+        },
+        "chartQueryResults": {
+          "type": "array",
+          "description": "Query results for table insights query.",
+              "items": {
+              "$ref": "#/definitions/InsightsTableResult",
+              "description": "Query results for table insights query."
+              }
+        },
+        "kind": {
+          "type": "string",
+          "description": "Insight or Activity type",
+          "enum": [
```

Comment:
I am not sure about the way we need to define enum as string. 
In some places I see diferent defenition, that include` x-ms-enum`

for example:
```
"enum": [
            "Scheduled",
            "MicrosoftSecurityIncidentCreation",
            "Fusion"
          ],
          "type": "string",
          "x-ms-enum": {
            "modelAsString": true,
            "name": "AlertRuleKind",
            "values": [
              {
                "value": "Scheduled"
              },
              {
                "value": "MicrosoftSecurityIncidentCreation"
              },
              {
                "value": "Fusion"
              }
            ]
          }
```

Reply:
[interesting point ](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-enum).
It seems that x-ms-enum are not enum, but string with documented recommended value, while enum is stricter.
It seems that using enum should be for API that rarely change, while x-ms-enum should be for list of values that often changed.